test/cri-containerd/container_test.go

```diff
@@ -594,3 +594,185 @@ func Test_RunContainer_NonDefault_User(t *testing.T) {
+
+	client := newTestRuntimeClient(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	sandboxRequest := getRunPodSandboxRequest(t, lcowRuntimeHandler)
+
+	podID := runPodSandbox(t, client, ctx, sandboxRequest)
+	defer removePodSandbox(t, client, ctx, podID)
+	defer stopPodSandbox(t, client, ctx, podID)
+
+	request := &runtime.CreateContainerRequest{
+		PodSandboxId: podID,
+		Config: &runtime.ContainerConfig{
+			Metadata: &runtime.ContainerMetadata{
+				Name: "",
+			},
+			Image: &runtime.ImageSpec{
+				Image: imageLcowAlpine,
+			},
+			Command: []string{
+				"top",
+			},
+			Annotations: map[string]string{
+				"containerd.io/snapshot/io.microsoft.container.storage.reuse-scratch": "true",
+			},
+		},
+		SandboxConfig: sandboxRequest.Config,
+	}
+
+	request.Config.Metadata.Name = "Container-1"
+	containerIDOne := createContainer(t, client, ctx, request)
+	defer removeContainer(t, client, ctx, containerIDOne)
+	startContainer(t, client, ctx, containerIDOne)
+	defer stopContainer(t, client, ctx, containerIDOne)
+
+	request.Config.Metadata.Name = "Container-2"
+	containerIDTwo := createContainer(t, client, ctx, request)
+	defer removeContainer(t, client, ctx, containerIDTwo)
+	startContainer(t, client, ctx, containerIDTwo)
+	defer stopContainer(t, client, ctx, containerIDTwo)
+
+	// Write a 10MB file and then check size available on the root fs in both containers.
+	// It should be the same amount as it's being backed by the same disk.
+	cmd := []string{"dd", "if=/dev/urandom", "of=testfile", "bs=10MB", "count=1"}
+	containerExecReq := &runtime.ExecSyncRequest{
+		ContainerId: containerIDOne,
+		Cmd:         cmd,
+		Timeout:     20,
+	}
+	r := execSync(t, client, ctx, containerExecReq)
+
+	if r.ExitCode != 0 {
+		t.Fatalf("failed with exit code %d: %s", r.ExitCode, string(r.Stderr))
+	}
+
+	// Grab the size available after the file write
+	cmd = []string{"df"}
+	containerExecReq = &runtime.ExecSyncRequest{
+		ContainerId: containerIDOne,
+		Cmd:         cmd,
+		Timeout:     20,
+	}
+	r = execSync(t, client, ctx, containerExecReq)
+
+	if r.ExitCode != 0 {
+		t.Fatalf("failed with exit code %d: %s", r.ExitCode, string(r.Stderr))
+	}
+
+	// Format of output for df is below
+	// Filesystem           1K-blocks      Used Available Use% Mounted on
+	// overlay               20642524        36  19577528   0% /
+	// tmpfs                    65536         0     65536   0% /dev
+	var (
+		scanner = bufio.NewScanner(strings.NewReader(string(r.Stdout)))
+		cols    []string
+		found   bool
+	)
+	for scanner.Scan() {
+		outputLine := scanner.Text()
+		if cols = strings.Fields(outputLine); cols[0] == "overlay" && cols[5] == "/" {
+			found = true
+			t.Log(outputLine)
+			break
+		}
+	}
+
+	if !found {
+		t.Fatalf("could not find the correct output line for overlay mount on / n: error: %v, exitcode: %d", string(r.Stdout), r.ExitCode)
+	}
+	availableSizeContainerOne := cols[3]
```

Comment:
nit: maybe make a helper out of this

Reply:
Yea we use this logic for another test also so good call